nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroFactoryService.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+/**
+ * This service allows to register factory classes and replacements.<br>
+ * <br>
+ * When a {@link AvroFactoryContext} is created factories are instanciated and passed to the context.
+ *
+ * @since 10.2
+ */
+public interface AvroFactoryService {
```

Comment:
Is this service limited to register Avro Schema factories ? will it be extended to encode/decode objects ?

Reply:
I planned to let it have this single responsibility and go for a third one.
Maybe later regroup things that are hardly depend but I'm not a big fan of tight coupling.